### DISCUSSION TITLE
libSQL & Turso database integration

### DISCUSSION THREAD

**Discussion by george-lim** [ory/kratos#3930] [Ideas]
Hey!

Wanted to start a discussion about supporting [Turso](https://turso.tech/) and its underlying database provider [libSQL](https://github.com/tursodatabase/libsql) for Ory Kratos. Turso has been gaining popularity recently, and I have been using it in a variety of products in productions.

My latest project involves pairing Ory Kratos with a self-hosted libSQL database. This would require me to connect to the database via the `libsql+ws://` scheme. I have chosen a tech stack that is mostly compatible with libSQL, except for Ory Kratos, which means I'll need a separate database for auth.

Does this sound like something that the team would consider adding support for? Happy to answer any questions surrounding libSQL (I am not affiliated with them in any way)

**Comment by sus-jordan** [comment]
Hi y'all. My startup company is also interested in a LibSQL database provider as a driver for our Ory ecosystem apps. There are a few questions I have for the community at large (or from the maintainers themselves) to further this discussion.

1. What are the differences between development and production mode data storage drivers, if any?
2. Where does compatibility end between the database driver for SQLite and the production driver, which is compatible with Postgres or MySQL?
3. Does LibSQL's HTTP-based connection model introduce problems for the ecosystem's SQL driver? If so, what problems would arise as a result of attempting to integrate a LibSQL provider into the existing driver?
4. Do the default SQLite plugins, which are loaded by all Turso databases meet the requirements for Ory ecosystem apps, like Kratos or Hydra?
